add backport api

cccclai opened the pull request:
```
(base) chenlai@chenlai-mp reuse_constant % python test_backport.py
[W export_module.cpp:685] Warning: The input bytecode model version is 4. Currently only support backporting to version 4 (function BackPortByteCode)
```

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56239 add backport api**
* #56002 [PyTorch Edge] Reuse constant table from ts in bytecode

cccclai committed:
add backport api

[ghstack-poisoned]

cccclai committed:
Update on "add backport api"




[ghstack-poisoned]

kimishpatel commented:
This is just exposing back porting API and not really doing any backporting. Is that right?

cccclai commented:
> This is just exposing back porting API and not really doing any backporting. Is that right?

Yeah this pr is not ready for review yet..

cccclai committed:
Update on "add backport api"




[ghstack-poisoned]

cccclai committed:
Update on "add backport api"

```
(base) chenlai@chenlai-mp reuse_constant % python test_backport.py
[W export_module.cpp:685] Warning: The input bytecode model version is 4. Currently only support backporting to version 4 (function BackPortByteCode)
```




[ghstack-poisoned]

cccclai commented:
deployed in pr https://github.com/pytorch/pytorch/pull/56802
